systemtest/aggregation_test.go

```diff
@@ -139,6 +139,11 @@ func TestTransactionAggregationShutdown(t *testing.T) {
 		estest.TermQuery{Field: "processor.event", Value: "transaction"},
 	)
 
+	// To avoid flaky tests, we wait for an additional 1s after the transaction
+	// has been indexed so the can aggregate the transaction before the server
+	// is stopped.
+	<-time.After(time.Second)
```

Comment:
Shutting down the server should cause the aggregator to flush: https://github.com/elastic/apm-server/blob/06699bac0cefe16833238622a632b8aa667d7699/x-pack/apm-server/aggregation/txmetrics/aggregator.go#L151

Seems like a bug if that's not happening?